From a07d1016002c20f4092442a47e4316d795db8928 Mon Sep 17 00:00:00 2001
From: Claudiu Manoil <claudiu.manoil@nxp.com>
Date: Wed, 16 Jun 2021 12:44:29 +0300
Subject: [PATCH 1/3] [AAVB-3196] pfeng: Enable pfe eth netdevs to be selected
 as DSA masters

The pfe netdev instances need to be discoverable via
of_find_net_device_by_node() so that the DSA code can
assign a such netdev instance connected to a switch as
a master port.
The issue is that there is only one platform device shared
by all netdev instances, the pfe ports don't have platform
devices linked to their corresponding "ethernet" DT nodes
of their own.  However of_find_net_device_by_node() allows
the netdev->dev to hold the device node link instead,
to cover for this case too.

Signed-off-by: Claudiu Manoil <claudiu.manoil@nxp.com>
---
 sw/linux-pfeng/pfeng-netif.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sw/linux-pfeng/pfeng-netif.c b/sw/linux-pfeng/pfeng-netif.c
index d9c6bfc42..d65dacb38 100644
--- a/sw/linux-pfeng/pfeng-netif.c
+++ b/sw/linux-pfeng/pfeng-netif.c
@@ -1011,6 +1011,7 @@ static struct pfeng_netif *pfeng_netif_logif_create(struct pfeng_priv *priv, str
 
 	/* Set the sysfs physical device reference for the network logical device */
 	SET_NETDEV_DEV(netdev, dev);
+	netdev->dev.of_node = netif_cfg->dn; /* required by of_find_net_device_by_node(), AAVB-3196 */
 
 	/* set ifconfig visible config */
 	netdev->mem_start = (unsigned long)priv->pfe_cfg->cbus_base;
-- 
2.32.0

